models/top_level/ServerConfig.py

```diff
@@ -70,11 +73,13 @@ def get_invalid_properties(self):
         """Checks the values of mandatory fields: bind (host), url, languages."""
         all_invalid_fields = []
 
-        if len(self.bind.host) < 7:
+        if not is_valid_string(self.bind.host):
             all_invalid_fields.append("server.bind.host")
-        if len(self.url) < 7:
-            all_invalid_fields.append("server.url")
         if len(self.languages) == 0:
             all_invalid_fields.append("server.languages")
+            
+        parsed_url = urlparse(self.url)
+        if not all([parsed_url.scheme, parsed_url.netloc]):
```

Comment:
The url can be literally an environmental variable, like in [this](https://github.com/byteroad/ogcapi-simple/blob/5f8e1b1a3753032fc12830871015143ba4f77e86/pygeoapi/docker.config.yml#L5C8-L5C19) example. It does not necessarily follow a url scheme, so you can leave it as a free form string.